src/core.jl

```diff
@@ -19,3 +19,157 @@ immutable ImageEdge
     index2::Int
     weight::Float64
 end
+
+"""
+    G, vert_map = region_adjacency_graph(seg, [weight_fn])
+
+Constructs a region adjacency graph from the `SegmentedImage`. It returns the RAG
+along with a label->vertex map. Optionally, a weight function `weight_fn` might be
+provided to set the edge weights. `weight_fn` takes two segment means and returns
+the weight of the connecting edge.
+
+"""
+function region_adjacency_graph{T<:SegmentedImage}(s::T, weight_fn::Function = default_diff_fn)
+
+    function neighbor_regions!{T<:SegmentedImage}(s::T, I::CartesianIndex, visited::AbstractArray, n::Set{Int})
```

Comment:
AFAICT `visited` and `n` are the inputs being modified, and that `n` is the "true output" of this function. Make `n` first and `visited` either second or last?